shell/renderer/api/atom_api_renderer_ipc.cc

```diff
@@ -124,13 +131,14 @@ class IPCRenderer : public gin::Wrappable<IPCRenderer> {
     if (!gin::ConvertFromV8(isolate, arguments, &message)) {
       return;
     }
-    electron_browser_ptr_->get()->MessageHost(channel, std::move(message));
+    electron_browser_ptr_->MessageHost(channel, std::move(message));
   }
 
   blink::CloneableMessage SendSync(v8::Isolate* isolate,
                                    bool internal,
                                    const std::string& channel,
                                    v8::Local<v8::Value> arguments) {
+    gin_helper::ErrorThrower thrower(isolate);
     blink::CloneableMessage message;
```

Comment:
check `mojo_connection_error_msg_` at the top here and throw.
```suggestion
    blink::CloneableMessage message;
    if (mojo_connection_encountered_error_) {
      thrower.ThrowError("IPC connection fatally interrupted.");
      return message;
    }
```

(and similarly for other methods here)